tests/DoctrineORMModuleTest/Service/DBALConnectionFactoryTest.php

```diff
@@ -0,0 +1,120 @@
+    {
+        $config = array(
+            'doctrine' => array(
+                'connection' => array(
+                    'orm_default' => array(
+                        'driverClass'   => 'Doctrine\DBAL\Driver\PDOSqlite\Driver',
+                        'params' => array(
+                            'memory' => true,
+                        ),
+                        'doctrineTypeMappings' => array(
+                            'money' => "string"
+                        ),
+                    )
+                ),
+            ),
+        );
+        $configurationMock = $this->getMockBuilder("Doctrine\ORM\Configuration")
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->serviceManager->setService('doctrine.configuration.orm_default', $configurationMock);
+        $this->serviceManager->setService('Config', $config);
+        $this->serviceManager->setService('Configuration', $config);
+        $dbal = $this->factory->createService($this->serviceManager);
+        $this->assertSame("string", $dbal->getDatabasePlatform()->getDoctrineTypeMapping("money"));
+    }
+
+    public function testDoctrineAddCustomCommentedType()
+    {
+        $config = array(
+            'doctrine' => array(
+                'connection' => array(
+                    'orm_default' => array(
+                        'driverClass'   => 'Doctrine\DBAL\Driver\PDOSqlite\Driver',
+                        'params' => array(
+                            'memory' => true,
+                        ),
+                        'doctrineTypeMappings' => array(
+                            'money' => 'money',
+                        ),
+                        'doctrineCommentedTypes' => array(
+                            'money'
+                        ),
+                    )
+                ),
+                'configuration' => array(
+                    'orm_default' => array(
+                        'types' => array(
+                            'money' => '\DoctrineORMModuleTest\Util\Type\MoneyType',
+                        ),
+                    ),
+                ),
+            ),
+        );
+        $configurationFactory = new ConfigurationFactory('orm_default');
+        $this->serviceManager->setService('Config', $config);
+        $this->serviceManager->setService('Configuration', $config);
+        $this->serviceManager->setService(
+            'doctrine.driver.orm_default',
+            $this->getMock('Doctrine\Common\Persistence\Mapping\Driver\MappingDriver')
+        );
+        $this->serviceManager->setService(
+            'doctrine.configuration.orm_default',
+            $configurationFactory->createService($this->serviceManager)
+        );
+        $dbal = $this->factory->createService($this->serviceManager);
+        $this->assertInstanceOf('DoctrineORMModuleTest\Util\Type\MoneyType', Type::getType($dbal->getDatabasePlatform()->getDoctrineTypeMapping("money")));
+    }
```

Comment:
This test is not verifying that the platform registered the DBAL type as a commented type